src/stream/producer.js

```diff
@@ -31,12 +31,384 @@
+  notification: 'notification',
+  failed: 'failed',
+  duplicate: 'duplicate'
+}
+/**
+ * The HTTP method for the process
+ *
+ * This ENUM is for the HTTP method that called the producer
+ *
+ * @typedef {object} ENUMS~METHOD
+ * @property {string} get
+ * @property {string} post
+ * @property {string} put
+ * @property {string} delete
+ */
+const METHOD = {
+  get: 'get',
+  post: 'post',
+  put: 'put',
+  del: 'delete'
+}
+/**
+ * The status of the process being posted to the topic
+ *
+ * This ENUM is for the STATUS of the message being produced
+ *
+ * @typedef {object} ENUMS~STATUS
+ * @property {string} success - successful validation passed
+ * @property {string} failed - failed validation
+ * @property {string} pending - pending proccessing
+ */
+const STATUS = {
+  success: 'success',
+  failure: 'failed',
+  pending: 'pending'
+}
+/**
+ * ENUMS
+ *
+ * Global ENUMS object
+ *
+ * @typedef {object} ENUMS
+ * @property {object} EVENTS - This ENUM is for the events
+ * @property {object} METHOD - This ENUM is for the METHOD
+ * @property {object} STATUS - This ENUM is for the STATUS
+ */
+const
+ENUMS = {
+  EVENTS,
+  METHOD,
+  STATUS
+}
+
+module.exports = ENUMS
+
+/**
+ * Producer class for adding messages to Kafka
+ *
+ * This is the main entry point for writing data to Kafka. You
+ * configure this like you do any other client, with a global
+ * configuration.
+ *
+ * @example
+ * var producer = new Producer(options, {
+ *   rdkafkaConf: {
+ *     'metadata.broker.list': options['metadata.broker.list'] || 'localhost:9092',
+ *     'client.id': options['client.id'] || 'default-client',
+ *     'event_cb': true,
+ *     'compression.codec': options['compression.codec'] || 'none',
+ *     'retry.backoff.ms': options['retry.backoff.ms'] || 100,
+ *     'message.send.max.retries': options['message.send.max.retries'] || 2,
+ *     'socket.keepalive.enable': options['socket.keepalive.enable'] || true,
+ *     'queue.buffering.max.messages': options['queue.buffering.max.messages'] || 10,
+ *     'queue.buffering.max.ms': options['queue.buffering.max.ms'] || 50,
+ *     'batch.num.messages': options['batch.num.messages'] || 10000,
+ *     'api.version.request': true,
+ *     'dr_cb': true
+ *   },
+ *   topicConf: {
+ *    'request.required.acks': options.requiredAcks || 1
+ *   }
+ * })
+ *
+ * @param {object} options - Key value pairs for mapping to the configuration
+ * @param {object} config - Key value pairs for the configuration of the Producer with the following:
+ * rdkafkaConf - specific rdkafka configurations [Refer to configuration doc]{@link https://github.com/edenhill/librdkafka/blob/0.11.1.x/CONFIGURATION.md}
+ * topicConf - topic configuration
+ * logger - logger object that supports debug(), info(), verbose(), error() & silly()
+ * @extends EventEmitter
+ * @constructor
+ */
+class Producer extends EventEmitter {
+  constructor (options = {requiredAcks: -1, partitionCount: 1, pollIntervalMs: 100}, config = {
```

Comment:
This does not conform to the standard config approach for the consumer

Reply:
changed